File: openrest4j/openrest4j-api/src/com/googlecode/openrest/v1_1/Category.java
package com.googlecode.openrest.v1_1;

import java.io.Serializable;
import java.util.Collections;
import java.util.List;
import java.util.Map;

import org.codehaus.jackson.annotate.JsonIgnoreProperties;
import org.codehaus.jackson.map.annotate.JsonSerialize;

@JsonIgnoreProperties(ignoreUnknown = true)
public class Category implements Serializable {
    /** Constructs a previously submitted category from persisted data. */
    public Category(String id, String restaurantId, Map<String, String> title,
    		Map<String, String> description, String parentCategoryId,
    		List<String> itemIds, Double priority) {
        this.id = id;
        this.restaurantId = restaurantId;
        this.title = title;
        this.description = description;
        this.parentCategoryId = parentCategoryId;
        this.itemIds = itemIds;
        this.priority = priority;
    }

    /** Constructs a new category to be submitted. */
    public Category(Map<String, String> title, Map<String, String> description,
    		String parentCategoryId, List<String> itemIds, Double priority) {
        this(null, null, title, description, parentCategoryId, itemIds, priority);
    }

    /** Default constructor for JSON deserialization. */
    public Category() {}

    /** The category's unique id. */
    @JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
    public String id;

    /** The restaurant's id. */
    @JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
    public String restaurantId;

    /** The category's title in various locales. */
    @JsonSerialize(include = JsonSerialize.Inclusion.NON_DEFAULT)
    public Map<String, String> title;

    /** The category's description in various locales. */
    @JsonSerialize(include = JsonSerialize.Inclusion.NON_DEFAULT)
    public Map<String, String> description;

    /** The parent category's id. */
    @JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
    public String parentCategoryId;

    /** The item-ids in this category. */
    @JsonSerialize(include = JsonSerialize.Inclusion.NON_DEFAULT)
    public List<String> itemIds = Collections.emptyList();

    /** Order priority. Lower numbers appear first in the list. */
    @JsonSerialize(include = JsonSerialize.Inclusion.NON_DEFAULT)
    public Double priority = 0.0;
    
    private static final long serialVersionUID = 1L;
}

File: trunk/openrest4js/openrest4js-api/Category.js
function CategoryFromObj(data)
{
    return new Category(data.id, data.restaurantId, data.description, data.itemIds, data.title, data.parentCategoryId, data.properties);
}

function Category(id, restaurantId, description, itemIds, title, parentCategoryId, properties)
{
    this.id = id;
    this.restaurantId = restaurantId;
    this.description = description;
    this.itemIds = itemIds;
    this.title = title;
    this.parentCategoryId = parentCategoryId;
    this.properties = ORDeserializeProperties(properties);
}

Category.prototype.toOpenRestObj = function()
{
    var ret = {};

    if (this.id) ret['id'] = this.id;
    if (this.restaurantId) ret['restaurantId'] = this.restaurantId;
    if (this.description) ret['description'] = this.description;
    if (this.itemIds) ret['itemIds'] = this.itemIds;
    if (this.title) ret['title'] = this.title;
    if (this.parentCategoryId) ret['parentCategoryId'] = this.parentCategoryId;
    if (this.properties) ret['properties'] = ORSerializeProperties(this.properties);

    return ret;
}
